test/shared/serviceClientTests.ts

```diff
@@ -0,0 +1,263 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import * as assert from "assert";
+import { getPropertyParent, PropertyParent, applyParameterTransformations } from "../../lib/serviceClient";
+import { OperationArguments } from "../../lib/operationArguments";
+import { Serializer } from "../../lib/serializer";
+
+describe("applyParameterTransformations()", () => {
+  it("should do nothing if provided an undefined operationArguments", () => {
+    applyParameterTransformations(
+      undefined as any,
```

Comment:
are there cases when we will suspect the generator of passing values like this?

Reply:
No, this should never happen. The only reason these tests are here is because I like to know how my code will behave even in these unexpected scenarios. I'm fine if the code throws an error, but I want to know what that error will be while I'm running my tests and not when my code is running on a customer's machine.